### PR TITLE
Auto detect bugs - take 3

### PR DESCRIPTION
**Pull request by rogeralsing**
I've improved the analyzer to not list types as "not used" if they occur in any of the config files, or are represented as strings in any of the other code files. 
Tests/Specs have also been removed from the list.

There are still quite a few classes that seems to have no purpose.

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Is this still in "do not merge" land?

**Comment by rogeralsing** [comment]
Yes, this is not a fix, this is just to highlight potential problems.
Why do we have unused types? What are their purpose?
They are a sign that something is wrong.

Maybe we should create individual issues from these?

**Comment by Aaronontheweb** [comment]
> They are a sign that something is wrong.

Not necessarily - some of the types like `DbSnapshotStore` are consumed downstream inside the Akka.Persistence.SqlServer and Postgres libraries, and `ClusterMetricsCollector` is going to be refactored into Akka.Cluster.Metrics

**Comment by rogeralsing** [comment]
That is what i mean, we have public api types with 0 coverage in the main repo.
How do we verify that the api is correct if we have nothing touching these types in here?
We could mess things up and not notice until another repo is built.

File: src/examples/Chat/ChatMessages/Messages.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public class ChannelsRequest
     {
     }
 
-    public class ChannelsResponse 
+    /*TODO: this class is not used*/public class ChannelsResponse
```
Comment (rogeralsing) [inline src/examples/Chat/ChatMessages/Messages.cs]: To be deleted

File: src/examples/Cluster/Samples.Cluster.Simple/SimpleClusterListener2.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 namespace Samples.Cluster.Simple
 {
-    public class SimpleClusterListener2 : UntypedActor
+    /*TODO: this class is not used*/public class SimpleClusterListener2 : UntypedActor
```
Comment (rogeralsing) [inline src/examples/Cluster/Samples.Cluster.Simple/SimpleClusterListener2.cs]: @Aaronontheweb is this used somewhere in the example?

File: src/examples/PersistenceExample/AtLeastOnceDeliveryExampleActor.cs [hunk 3]
```diff
@@ -43,7 +43,7 @@ public Confirmation(long deliveryId)
 
         public long DeliveryId { get; private set; }
     }
-    [Serializable]
+    /*TODO: this class is not used*/[Serializable]
```
Comment (rogeralsing) [inline src/examples/PersistenceExample/AtLeastOnceDeliveryExampleActor.cs]: @Horusiath is this used somewhere in the example?

File: src/core/Akka/Routing/ConsistentHash.cs [hunk 4]
```diff
@@ -134,7 +134,7 @@ public bool IsEmpty
             get { return !_nodes.Any(); }
         }
         
-        public class ConsistentHashingGroupSurrogate : ISurrogate
+        /*TODO: this class is not used*/public class ConsistentHashingGroupSurrogate : ISurrogate
```
Comment (rogeralsing) [inline src/core/Akka/Routing/ConsistentHash.cs]: Analyzer bug , this is used

File: src/core/Akka/IO/Tcp.cs [hunk 5]
```diff
@@ -762,7 +762,7 @@ internal MessageDispatcher FileIoDispatcher
         }
     }
 
-    public class TcpMessage
+    /*TODO: this class is not used*/public class TcpMessage
```
Comment (rogeralsing) [inline src/core/Akka/IO/Tcp.cs]: @willieferguson is this used? Seen a few IO classes that seems to have 0 refs

File: src/core/Akka/Dispatch/SysMsg/ISystemMessage.cs [hunk 6]
```diff
@@ -27,7 +27,7 @@ public interface ISystemMessage : INoSerializationVerificationNeeded
     /// <summary>
     ///     Class NoMessage.
     /// </summary>
-    public sealed class NoMessage : ISystemMessage
+    /*TODO: this class is not used*/public sealed class NoMessage : ISystemMessage
```
Comment (rogeralsing) [inline src/core/Akka/Dispatch/SysMsg/ISystemMessage.cs]: We do not use this one yet, but it is used in the mailbox impl in scala. We need to look into its purpose

File: src/core/Akka/Actor/Stash/UntypedActorWithBoundedStash.cs [hunk 7]
```diff
@@ -13,7 +13,7 @@ namespace Akka.Actor
     /// <summary>
     /// An UntypedActor with bounded Stash capabilities
     /// </summary>
-    [Obsolete("Inherit from UntypedActor and IWithBoundedStash instead. This class will be removed in a future release.")]
+    /*TODO: this class is not used*/[Obsolete("Inherit from UntypedActor and IWithBoundedStash instead. This class will be removed in a future release.")]
```
Comment (rogeralsing) [inline src/core/Akka/Actor/Stash/UntypedActorWithBoundedStash.cs]: Obsoleted, at what point should we remove obsoleted?

File: src/core/Akka/Actor/Props.cs [hunk 8]
```diff
@@ -567,7 +567,7 @@ public void Release(ActorBase actor)
             }
         }
 
-        private class FactoryConsumer<TActor> : IIndirectActorProducer where TActor : ActorBase
+        /*TODO: this class is not used*/private class FactoryConsumer<TActor> : IIndirectActorProducer where TActor : ActorBase
```
Comment (rogeralsing) [inline src/core/Akka/Actor/Props.cs]: Anyone know what this is used for? @akkadotnet

File: src/contrib/loggers/Akka.Logger.slf4net/Slf4NetLogger.cs [hunk 9]
```diff
@@ -18,7 +18,7 @@ namespace Akka.Logger.slf4net
     /// recognized: <see cref="Debug"/>, <see cref="Info"/>,
     /// <see cref="Warning"/> and <see cref="Error"/>.
     /// </summary>
-    public class Slf4NetLogger : UntypedActor
+    /*TODO: this class is not used*/public class Slf4NetLogger : UntypedActor
```
Comment (rogeralsing) [inline src/contrib/loggers/Akka.Logger.slf4net/Slf4NetLogger.cs]: Used via config, logger, add some tests for this

File: src/core/Akka.Persistence/JournalProtocol.cs [hunk 10]
```diff
@@ -38,7 +38,7 @@ public DeleteMessagesSuccess(IEnumerable<IPersistentEnvelope> messageIds)
     /// <summary>
     /// Reply message to failed <see cref="DeleteMessages"/> request.
     /// </summary>
-    public sealed class DeleteMessagesFailure
+    /*TODO: this class is not used*/public sealed class DeleteMessagesFailure
```
Comment (rogeralsing) [inline src/core/Akka.Persistence/JournalProtocol.cs]: @Horusiath , is this used? No refs to it

File: src/core/Akka.Remote/Endpoint.cs [hunk 11]
```diff
@@ -233,7 +233,7 @@ public EndpointAssociationException(string msg)
     /// <summary>
     /// INTERNAL API
     /// </summary>
-    internal sealed class OversizedPayloadException : EndpointException
+    /*TODO: this class is not used*/internal sealed class OversizedPayloadException : EndpointException
```
Comment (rogeralsing) [inline src/core/Akka.Remote/Endpoint.cs]: @Aaronontheweb is or should this be used?

File: src/core/Akka.Remote/Transport/AkkaProtocolTransport.cs [hunk 12]
```diff
@@ -327,7 +327,7 @@ internal enum AssociationState
 
     internal class HeartbeatTimer : INoSerializationVerificationNeeded { }
 
-    internal sealed class HandleMsg : INoSerializationVerificationNeeded
+    /*TODO: this class is not used*/internal sealed class HandleMsg : INoSerializationVerificationNeeded
```
Comment (rogeralsing) [inline src/core/Akka.Remote/Transport/AkkaProtocolTransport.cs]: @Aaronontheweb what is this for?

File: src/core/Akka/Actor/BuiltInActors.cs [hunk 13]
```diff
@@ -15,7 +15,7 @@ namespace Akka.Actor
     /// <summary>
     ///     Class EventStreamActor.
     /// </summary>
-    public class EventStreamActor : ActorBase
+    /*TODO: this class is not used*/public class EventStreamActor : ActorBase
```
Comment (rogeralsing) [inline src/core/Akka/Actor/BuiltInActors.cs]: Anyone? Just an unfinished impl? Is it a leftover from the early days? 
Im feeling guilty but dont know if I did this :)

File: src/core/Akka/Actor/FSM.cs [hunk 14]
```diff
@@ -208,7 +208,7 @@ public void Cancel()
         /// </summary>
         /// <typeparam name="TS">The name of the state</typeparam>
         /// <typeparam name="TD">The data of the state</typeparam>
-        public class LogEntry<TS, TD>
+        /*TODO: this class is not used*/public class LogEntry<TS, TD>
```
Comment (rogeralsing) [inline src/core/Akka/Actor/FSM.cs]: @Aaronontheweb is this used?

File: src/core/Akka/IO/Inet.cs [hunk 15]
```diff
@@ -106,7 +106,7 @@ public override void AfterConnect(Socket s)
                 }
             }
 
-            public class TrafficClass : SocketOption
+            /*TODO: this class is not used*/public class TrafficClass : SocketOption
```
Comment (rogeralsing) [inline src/core/Akka/IO/Inet.cs]: @willieferguson are these used?

File: src/core/Akka/IO/Tcp.cs (hunk 5 above)
Comment (willieferguson) [inline src/core/Akka/IO/Tcp.cs]: These are factory methods which can be used by a Akka IO client. In JVM Akka this is for the Java API. I generally just used the message constructor:
new Tcp.Connect(...)  vs
TcpMessage.Connect(..)

I suppose we can either remove the TcpMessage factory class or keep it and make the message constructors internal.

File: src/core/Akka/IO/Inet.cs (hunk 15 above)
Comment (willieferguson) [inline src/core/Akka/IO/Inet.cs]: There is an outstanding TODO in that class to investigate .NET equivalent and port.

File: src/core/Akka.Remote/Transport/AkkaProtocolTransport.cs (hunk 12 above)
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/AkkaProtocolTransport.cs]: Good question - not sure offhand anymore.

File: src/core/Akka.Remote/Endpoint.cs (hunk 11 above)
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Endpoint.cs]: This should be used - I have an outstanding issue #879 related to this,

File: src/core/Akka/Actor/Stash/UntypedActorWithBoundedStash.cs (hunk 7 above)
Comment (Aaronontheweb) [inline src/core/Akka/Actor/Stash/UntypedActorWithBoundedStash.cs]: 2.0

File: src/examples/Cluster/Samples.Cluster.Simple/SimpleClusterListener2.cs (hunk 2 above)
Comment (Aaronontheweb) [inline src/examples/Cluster/Samples.Cluster.Simple/SimpleClusterListener2.cs]: No, it can be removed.

File: src/core/Akka/Actor/FSM.cs (hunk 14 above)
Comment (Aaronontheweb) [inline src/core/Akka/Actor/FSM.cs]: Meant for the `LoggingFSM` - I'm not sure if that's ever been finished. Basically it's an `FSM` implementation that logs its transitions.

**Comment by rogeralsing** [comment]
This can be merged.
Some classes have been removed, some small closeover fixes, and some more TODOs.

**Pull request closed without merging**